math/vector_cross_product.cpp

```diff
@@ -0,0 +1,111 @@
+/**
+ * @file
+ *
+ * @brief Calculates the [Cross Product](https://en.wikipedia.org/wiki/Cross_product) and the magnitude of two mathematical vectors from their direction ratios.
+ *
+ *
+ * @details Cross Product of two vectors gives a vector.
+ * Direction Ratios of a vector are the numeric parts of the given vector. They are the tree parts of the
+ * vector which determine the magnitude (value) of the vector.
+ * The method of finding a cross product is the same as finding the determinant of an order 3 matrix consisting
+ * of the first row with unit vectors of magnitude 1, the second row with the direction ratios of the
+ * first vector and the third row with the direction ratios of the second vector.
+ * The magnitude of a vector is it's value expressed as a number.
+ * Let the direction ratios of the first vector, P be: a, b, c
+ * Let the direction ratios of the second vector, Q be: x, y, z
+ * Therefore the calculation for the cross product can be arranged as:
+ * P x Q:
+ *  	1	1	1
+ *  	a	b	c
+ *  	x	y	x
+ *
+ * @algorithm The direction ratios (DR) are calculated as follows:
+ *  	1st DR, J:  (b * x) - (c * y)
+ *  	2nd DR, A: -((a * x) - (c * x))
+ *  	3rd DR, N:  (a * y) - (b * x)
+ *
+ * Therefore, the direction ratios of the cross product are: J, A, N
+ * The following C++ Program calculates the direction ratios of the cross products of two vector.
+ * The program uses a function, cross() for doing so.
+ * The direction ratios for the first and the second vector has to be passed one by one seperated by a space character.
+ *
+ * Magnitude of a vector is the square root of the sum of the squares of the direction ratios.
+ *
+ * @example An example of a running instance of the executable program:
+ *
+ * 	Pass the first Vector: 1 2 3
+ *	Pass the second Vector: 4 5 6
+ *	The cross product is: -3 6 -3
+ *	Magnitude: 7.34847
+ *
+ * @author [Shreyas Sable](https://github.com/Shreyas-OwO)
+ */
+
+#include <iostream>
+#include <array>
+#include <cmath>
+
+namespace math {
+	/**
+	 * @namespace math
+	 * @brief Math algorithms
+	 */
+	namespace vector_cross {
+		/**
+		 * @namespace vector_cross
+		 * @brief Functions for Vector Cross Product algorithms
+		 */
+
+		/**
+		 * @brief Function to calculate the cross product of the passed arrays containing the direction ratios of the two mathematical vectors.
+		 * @param A type: std::array<double, 3> description: contains the direction ratios of the first mathematical vector.
```

Comment:
the typename is not specified in C/C++ docs
```suggestion
		 * @param A contains the direction ratios of the first mathematical vector.
```
please update the format throughout the code